Masterbar: Update the Reader link

ramonjd opened the pull request:
Fixes N/A

#### Changes proposed in this Pull Request:
We're updating the masterbar Reader link to explicitly go to `wordpress.com/read`

For background context, we'd like to set up an A/B test to redirect a (logged-in) test group to Customer Home instead of the Reader upon visiting `wordpress.com`

In order to achieve that we need to ensure we can manipulate the default landing page for `/`. Currently it is `/read`.

By updating instances of `/` to `/read`, we'll ensure that links to the Reader work regardless of which A/B test cohort the clicker is in.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
For context see:
* p58i-89H-p2
* EPIC Issue: https://github.com/Automattic/zelda-private/issues/173
* Calypso PR: https://github.com/Automattic/wp-calypso/pull/37547

#### Testing instructions
* Connect your local ngroked Jetpack application to a wordpress.com site
* Visit `http://{yoursiteurl}/wp-admin/plugins.php?calypsoify=1`
* The Reader link in the masterbar should link to https://wordpress.com/read

<img width="867" alt="Screen Shot 2019-12-12 at 2 50 05 pm" src="https://user-images.githubusercontent.com/6458278/70681733-7c381380-1cf0-11ea-8c79-fd8779791d0f.png">

* Now enable the WordPress.com toolbar
<img width="764" alt="Screen Shot 2019-12-13 at 11 34 49 am" src="https://user-images.githubusercontent.com/6458278/70760563-215cf580-1d9e-11ea-8f8b-6ecbea2781e2.png">

* The Followed Sites link in the sidebar should link to https://wordpress.com/read
<img width="499" alt="Screen Shot 2019-12-13 at 11 38 24 am" src="https://user-images.githubusercontent.com/6458278/70760564-215cf580-1d9e-11ea-82f1-ba4e3f04c7dc.png">


#### Proposed changelog entry for your changes:

* Updated the link to the Reader in the masterbar to point to https://wordpress.com/read

ramonjd committed:
Update the Reader link to go to wordpress.com/read as we're performing A/B tests in Calypso on different default landing page destinations. Context: https://github.com/Automattic/wp-calypso/pull/37547

andrewserong reviewed (dismissed):
Thanks for working on this one @ramonjd! Tested in my local Jetpack dev environment, and the masterbar Reader button now links through to `https://wordpress.com/read`.

Still might just want a review from @Automattic/jetpack-crew too, to see about getting this in the next release :)

kbrown9 reviewed (dismissed):
This looks good to me! I tested the change, and the Reader link in the masterbar links to https://wordpress.com/read.

jeherve reviewed (commented):
This works for me, but I wonder if we shouldn't also change the Reader link in the regular Masterbar, displayed to everyone even when Calypsoify is off:
https://github.com/Automattic/jetpack/blob/cd172084d26caf6111d867975f6e4bc878c61a2a/modules/masterbar/masterbar.php#L407

Or do we want those folks to keep using the old link?

ramonjd commented:
> This works for me, but I wonder if we shouldn't also change the Reader link in the regular Masterbar, displayed to everyone even when Calypsoify is off:

Thanks @jeherve  Totally missed that. All links to the Reader, whose underlying URL is wordpress.com should be switched to wordpress.com/read. 

I'll fix it up tomorrow.

jeherve committed:
Update Reader link in Masterbar as well

jeherve reviewed (approved):
Thanks for the confirmation! I updated the link for you, this should now be good to merge!

ramonjd commented:
> Thanks for the confirmation! I updated the link for you, this should now be good to merge!

Thanks @jeherve ! I updated the test instructions anyway since it took me a while to work out where I should be seeing this link :)
